File: DSA Practice/LeetCode/326. Power of Three/solution.py
class Solution:
    def isPowerOfThree(self, n: int) -> bool:
        
        def rec(n):
            if n == 1:
                return True

            elif n < 1:
                return False

            if n % 3 == 0:
                return rec(n/3)

            else :
                return False

        return rec(n)

File: DSA Practice/LeetCode/326. Power of Three/solution.java
class Solution {
    public boolean isPowerOfThree(int n) {
          return rec(n);
    }

    private boolean rec(int n){
        if (n == 1){
            return true;
        } else if (n < 1){
            return false;
        }

        if (n%3 == 0){
            return rec(n/3);
        } else {
            return false;
        }
    } 
}